Content.Shared/_Moffstation/Vampire/Components/BloodConsumptionComponent.cs

```diff
@@ -0,0 +1,68 @@
+﻿using Content.Shared.Chemistry.Reagent;
+using Content.Shared.Damage;
+using Robust.Shared.GameStates;
+using Robust.Shared.Prototypes;
+
+namespace Content.Shared._Moffstation.Vampire.Components;
+
+/// <summary>
+/// This is the component given to vampires and other creatures which utilize blood as their main form of sustenance.
+/// Interfacing largely with the entity's bloodstream, this component treats their bloodstream as their hunger value, and
+/// main form of healing.
+/// </summary>
+[RegisterComponent, NetworkedComponent]
+public sealed partial class BloodConsumptionComponent : Component
```

Comment:
By the way, I want to say I really like how you have done the tick update stuff. So many other implementations I've seen are really unclear with how often they update, and how the "ChangeInWhatevers" relate to the update frequency.